File: qian/src/api/college/collegeInfo.js
import request from '@/utils/request'

// 获取高校信息列表
export function listCollegeInfo(query) {
    return request({
        url: '/api/college_info/',
        method: 'get',
        params: query
    })
}

// 获取高校信息详情
export function getCollegeInfo(id) {
    return request({
        url: '/api/college_info/' + id,
        method: 'get'
    })
}

// 新增高校信息
export function addCollegeInfo(data) {
    return request({
        url: '/api/college_info/',
        method: 'post',
        data: data
    })
}

// 修改高校信息
export function updateCollegeInfo(id, data) {
    return request({
        url: '/api/college_info/' + id,
        method: 'put',
        data: data
    })
}

// 删除高校信息
export function delCollegeInfo(id) {
    return request({
        url: '/api/college_info/' + id,
        method: 'delete'
    })
}

// 获取生源看板数据
export function getStudentSource(query) {
    return request({
        url: '/api/college_student_source/',
        method: 'get',
        params: query
    })
}

File: web/controller/college_info.py
from flask import Blueprint, request, jsonify
from web.models.college_info import CollegeInfo
from web import db
from web.decorator.permission import requires_roles, jwt_required

college_info_bp = Blueprint('college_info', __name__, url_prefix='/api/college_info')

@college_info_bp.route('/', methods=['GET'])
def list_college_info():
    page = request.args.get('page', 1, type=int)
    per_page = request.args.get('pageSize', 10, type=int)
    pagination = CollegeInfo.query.paginate(page=page, per_page=per_page, error_out=False)
    data = [item.to_dict() for item in pagination.items]
    return jsonify({"code": 0, "msg": "成功", "data": {"rows": data, "total": pagination.total}})

@college_info_bp.route('/<int:id>', methods=['GET'])
def get_college_info(id):
    info = CollegeInfo.query.get(id)
    if not info:
        return jsonify({"code": 1, "msg": "未找到"}), 404
    return jsonify({"code": 0, "msg": "成功", "data": info.to_dict()})

@college_info_bp.route('/', methods=['POST'])
@jwt_required
@requires_roles('admin')
def create_college_info():
    data = request.get_json()
    info = CollegeInfo(
        name=data.get('name'),
        intro=data.get('intro'),
        photo=data.get('photo'),
        specialties=data.get('specialties')
    )
    db.session.add(info)
    db.session.commit()
    return jsonify({"code": 0, "msg": "创建成功", "data": info.to_dict()})

@college_info_bp.route('/<int:id>', methods=['PUT'])
@jwt_required
@requires_roles('admin')
def update_college_info(id):
    info = CollegeInfo.query.get(id)
    if not info:
        return jsonify({"code": 1, "msg": "未找到"}), 404
    data = request.get_json()
    for field in ['name', 'intro', 'photo', 'specialties']:
        if field in data:
            setattr(info, field, data[field])
    db.session.commit()
    return jsonify({"code": 0, "msg": "更新成功", "data": info.to_dict()})

@college_info_bp.route('/<int:id>', methods=['DELETE'])
@jwt_required
@requires_roles('admin')
def delete_college_info(id):
    info = CollegeInfo.query.get(id)
    if not info:
        return jsonify({"code": 1, "msg": "未找到"}), 404
    db.session.delete(info)
    db.session.commit()
    return jsonify({"code": 0, "msg": "删除成功"})
